src/posit/connect/resources.py

```diff
@@ -3,11 +3,18 @@
 from abc import ABC, abstractmethod
 from typing import Generic, List, Optional, TypeVar
 
+import requests
+
 
 T = TypeVar("T")
 
 
 class Resource(ABC, dict):
+    def __init__(self, session: requests.Session, url: str, **kwargs):
+        self._session = session
+        self._url = url
```

Comment:
`__setattr__` was blocked because we want ambiguity in whether the object we have locally is in sync with the server, and we should be clear when an action will result in a network request. Routing everything through `.update()` makes that explicit.

I think you can retain the `__setattr__` (which belongs in the `Resource` class, not `User`) if you do `super().__setattr__("session", session)` here.

Unrelated comment to that: can we keep them as `session` and `url` without the `_` prefix? We should be consistent across all of our objects.

Reply:
That works! Thank you!